lib/tbot/cli/start_workload_identity_x509.go

```diff
@@ -0,0 +1,110 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cli
+
+import (
+	"fmt"
+	"log/slog"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/tbot/config"
+)
+
+// WorkloadIdentityX509Command implements `tbot start workload-identity-x509` and
+// `tbot configure spiffe-svid`.
+type WorkloadIdentityX509Command struct {
+	*sharedStartArgs
+	*sharedDestinationArgs
+	*genericMutatorHandler
+
+	IncludeFederatedTrustBundles bool
+	// WorkloadIdentityName is the name of the workload identity to use.
+	// --workload-identity-name foo
+	WorkloadIdentityName string
+	// WorkloadIdentityLabels is the labels of the workload identity to use.
+	// --workload-identity-labels x=y,z=a
+	WorkloadIdentityLabels string
+}
+
+// NewWorkloadIdentityX509Command initializes the command and flags for the
+// `workload-identity-x509` output and returns a struct that will contain the parse
+// result.
+func NewWorkloadIdentityX509Command(parentCmd *kingpin.CmdClause, action MutatorAction, mode CommandMode) *WorkloadIdentityX509Command {
+	// TODO(noah): Unhide this command when feature flag removed
+	cmd := parentCmd.Command("workload-identity-x509", fmt.Sprintf("%s tbot with a SPIFFE-compatible SVID output.", mode)).Hidden()
+
+	c := &WorkloadIdentityX509Command{}
+	c.sharedStartArgs = newSharedStartArgs(cmd)
+	c.sharedDestinationArgs = newSharedDestinationArgs(cmd)
+	c.genericMutatorHandler = newGenericMutatorHandler(cmd, c, action)
+
+	cmd.Flag(
+		"include-federated-trust-bundles",
+		"If set, include federated trust bundles in the output",
+	).BoolVar(&c.IncludeFederatedTrustBundles)
+	cmd.Flag(
+		"workload-identity-name",
+		"The name of the workload identity to issue",
+	).StringVar(&c.WorkloadIdentityName)
+	cmd.Flag(
+		"workload-identity-labels",
```

Comment:
Is `--workload-identity-name` and `--workload-identity-labels` too verbose? I was worried that just `--name` / `--labels` would be not verbose enough. Perhaps `--name-selector` / `--label-selector` ?

Reply:
I'm actually curious about this exact question on https://github.com/gravitational/teleport/pull/50898. I've so far gone with the unambiguous but way too verbose names following the pattern here, but was strongly considering the much shorter `--name` and `--labels` since users can probably infer meaning from context. Maybe there's some additional benefit knowing that e.g. `--labels` can be reused across certain commands that accept selectors?

Reply:
Hm yeah - we'd want to switch both to use the same implementation of a label selector for sure. I do feel that `--name` is a little too generic. Perhaps `--name-selector`/`--label-selector` is the best middleground?

Reply:
Hmm, that seems like a good compromise. The only other option that comes to mind is `--select-name` if we want something less passive.

Reply:
0307c4750e